apps/meteor/app/oembed/server/server.ts

```diff
@@ -128,6 +135,36 @@ const getUrlContent = async (urlObj: URL, redirectCount = 5): Promise<OEmbedUrlC
 	};
 };
 
+const parseUrl = async function (url: string): Promise<{ urlPreview: MessageUrl; foundMeta: boolean }> {
+	const parsedUrlObject: MessageUrl = { url, meta: {} };
+	let foundMeta = false;
+	if (!isURL(url)) {
+		return { urlPreview: parsedUrlObject, foundMeta };
+	}
+
+	const data = await getUrlMetaWithCache(url);
+
+	if (data) {
+		if (isOEmbedUrlWithMetadata(data) && data.meta) {
+			parsedUrlObject.meta = getRelevantMetaTags(data.meta) || {};
+			if (parsedUrlObject.meta?.oembedHtml) {
+				parsedUrlObject.meta.oembedHtml = insertMaxWidthInOembedHtml(parsedUrlObject.meta.oembedHtml) || '';
+			}
+		}
+
+		if (data.headers?.contentLength) {
+			parsedUrlObject.headers = { ...parsedUrlObject.headers, contentLength: data.headers.contentLength };
+		}
+
+		if (data.headers?.contentType) {
+			parsedUrlObject.headers = { ...parsedUrlObject.headers, contentType: data.headers.contentType };
```

Comment:
even though this code was just moved from its previous location to here, but I'd say it could be refactored to be easier to read and don't do multiple spreads of `parsedUrlObject.headers` .. something like:


```ts
if (!isURL) {
  return ...;
}

if (!data) {
  return ...;
}

return {
   headers: {
    ...parsedUrlObject.headers,
    ...(data.headers?.contentLength ? { contentLength: data.headers.contentLength } : {}),
    ...(data.headers?.contentType ? { contentType: data.headers.contentType } : {}),
  }
}
```

wdyt?

Reply:
Good idea. I changed the final return statement to be more clear as you suggested
I had to add some extra checks, but it definitely looks clearer